ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py

```diff
@@ -0,0 +1,332 @@
+	if frappe.db.get_value("Company", company, "enable_perpetual_inventory"):
+		frappe.db.set_value("Company", company, "enable_perpetual_inventory", 0)
+
+
+def make_error_log(error_logs):
+	msg = "\n\n".join(err for err in error_logs)
+	log = frappe.new_doc("Zenoti Error Logs")
+	log.title = _("Errors occured at {}").format(get_datetime())
+	log.error_message = msg
+	log.insert()
+
+
+def add_genders():
+	for gender in ["NotSpecified", "Any", "ThirdGender", "Multiple"]:
+		if not frappe.db.exists("Gender", gender):
+			doc = frappe.new_doc("Gender")
+			doc.gender = gender
+			doc.insert()
+
+
+def make_item_group():
+	for item_group in ["Gift or Pre-paid Cards", "Memberships", "Packages"]:
+		if not frappe.db.exists("Item Group", item_group):
+			doc = frappe.new_doc("Item Group")
+			doc.item_group_name = item_group
+			doc.parent_item_group = "All Item Groups"
+			doc.insert()
+
+
+def make_item_tips():
+	if not frappe.db.exists("Item", "Tips"):
+		item = frappe.new_doc("Item")
+		item.item_code = "Tips"
+		item.item_name = "Tips"
+		item.item_group = "All Item Groups"
+		item.is_stock_item = 0
+		item.include_item_in_manufacturing = 0
+		item.stock_uom = "Nos"
+		item.insert()
+
+
+def setup_custom_fields():
+	custom_fields = {
+		"Supplier": [
+			dict(
+				fieldname="zenoti_supplier_code",
+				label="Zenoti Supplier Code",
+				fieldtype="Data",
+				insert_after="naming_series",
+				read_only=1,
+				print_hide=1,
+			)
+		],
+		"Customer": [
+			dict(
+				fieldname="zenoti_guest_id",
+				label="Zenoti Guest Id",
+				fieldtype="Data",
+				insert_after="salutation",
+				read_only=1,
+				print_hide=1,
+				hidden=1,
+			),
+			dict(
+				fieldname="zenoti_guest_code",
+				label="Zenoti Guest Code",
+				fieldtype="Data",
+				insert_after="zenoti_guest_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Item": [
+			dict(
+				fieldname="zenoti_item_id",
+				label="Zenoti Item Id",
+				fieldtype="Data",
+				insert_after="item_code",
+				read_only=1,
+				print_hide=1,
+				hidden=1,
+			),
+			dict(
+				fieldname="zenoti_item_category",
+				label="Zenoti Item Category",
+				fieldtype="Data",
+				insert_after="item_group",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_item_sub_category",
+				label="Zenoti Item Sub Category",
+				fieldtype="Data",
+				insert_after="zenoti_item_category",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_bussiness_unit_id",
+				label="Zenoti Bussiness Unit Id",
+				fieldtype="Data",
+				insert_after="zenoti_item_category",
+				read_only=1,
+				print_hide=1,
+			),
+			dict(
+				fieldname="zenoti_item_type",
+				label="Zenoti Item Type",
+				fieldtype="Select",
+				options="\nRetail\nConsumable\nBoth",
+				insert_after="zenoti_bussiness_unit_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Sales Invoice": [
+			dict(
+				fieldname="zenoti_invoice_no",
+				label="Zenoti Invoice No",
+				fieldtype="Data",
```

Comment:
```suggestion
				fieldtype="Data",
```

Sales Invoice doctype is hitting row limits when many integrations are active... is it possible to use `Small Text` for this or `receipt no` field down below?


I've changed this in other integrations (shopify/unicommerce)